tensorflow/compiler/xla/service/gpu/cudnn_conv_pad_for_integer_convolutions.cc

```diff
@@ -0,0 +1,129 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/xla/service/gpu/cudnn_conv_pad_for_integer_convolutions.h"
+
+#include "tensorflow/compiler/xla/literal_util.h"
+#include "tensorflow/compiler/xla/service/gpu/cudnn_conv_pad_features.h"
+#include "tensorflow/compiler/xla/service/gpu/ir_emission_utils.h"
+#include "tensorflow/compiler/xla/service/hlo_casting_utils.h"
+#include "tensorflow/compiler/xla/util.h"
+#include "tensorflow/compiler/xla/window_util.h"
+
+namespace xla {
+namespace gpu {
+
+// We always pad regardless the memory usage increase.
+// We still check the increase for debugging purpose.
+static constexpr double kMaxBytesTouchedIncrease = 1.35;
+
+static StatusOr<bool> ResolvePadedShapes(
```

Comment:
This function can be replaced with a lambda in theory.  But it is difficult to infer the correct return data type from a lambda in this case (returning true/false, while declaring return type as StatusOr<bool>).